pkg/controller/statefulset/stateful_set_utils_test.go

```diff
@@ -287,6 +288,26 @@ func TestNewPodControllerRef(t *testing.T) {
 	}
 }
 
+func TestCreateApplyRevision(t *testing.T) {
+	set := newStatefulSet(1)
+	revision, err := newRevision(set, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	set.Spec.Template.Spec.Containers[0].Name = "foo"
+	restored, err := applyRevision(set, revision)
+	if err != nil {
+		t.Fatal(err)
+	}
+	restoredRevision, err := newRevision(restored, 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !history.EqualRevision(revision, restoredRevision) {
```

Comment:
Yes the revisions are only compared by hash and by contents. Two revisions with different ".Revisions" can be equal using this comparison.